src/test/java/ch/wisv/areafiftylan/ConsumptionTest.java

```diff
@@ -0,0 +1,424 @@
+    }
+
+    @Test
+    public void getAllPossibleConsumptionsTestNone_NoAdmin(){
+        SessionData session = login(user.getUsername(), userCleartextPassword);
+
+        getAllPossibleTickets(session).
+            then().
+                statusCode(HttpStatus.SC_FORBIDDEN);
+    }
+
+    @Test
+    public void getAllPossibleConsumptionsTestNone_Admin(){
+        consumptionService.removePossibleConsumption(spicyFood.getId());
+        consumptionService.removePossibleConsumption(coldMilkshake.getId());
+
+
+        getAllPossibleTickets_AsAdmin().
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("$", hasSize(0));
+    }
+
+    @Test
+    public void getAllPossibleConsumptionsTestSingle_Admin(){
+        consumptionService.removePossibleConsumption(spicyFood.getId());
+
+        getAllPossibleTickets_AsAdmin().
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("$", hasSize(1)).
+                body("name", containsInAnyOrder(coldMilkshake.getName()));
+    }
+
+    @Test
+    public void getAllPossibleConsumptionsTestMultiple_Admin(){
+        getAllPossibleTickets_AsAdmin().
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("$", hasSize(2)).
+                body("name", containsInAnyOrder(coldMilkshake.getName(), spicyFood.getName()));
+    }
+
+    @Test
+    public void getIsConsumed_True(){
+        consumptionService.consume(ticket.getId(), spicyFood.getId());
+
+        getIsConsumed(ticket, spicyFood).
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("consumed", is(true));
+    }
+
+    @Test
+    public void getIsConsumed_False(){
+        getIsConsumed(ticket, spicyFood).
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("consumed", is(false));
+    }
+
+    @Test
+    public void getIsConsumed_ConsumptionDoesntExist(){
+        getIsConsumed(ticket.getId(), getUnusedConsumptionId()).
+                then().statusCode(HttpStatus.SC_NOT_FOUND);
+    }
+
+    @Test
+    public void getIsConsumed_InvalidTicket(){
+        invalidateTicket();
+
+        getIsConsumed(ticket, spicyFood).
+                then().statusCode(HttpStatus.SC_BAD_REQUEST);
+    }
+
+    @Test
+    public void getTicketConsumptions_NoneConsumed(){
+        getTicketConsumptions(ticket).
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("$", hasSize(0));
+    }
+
+    @Test
+    public void getTicketConsumptions_OneConsumed(){
+        consumptionService.consume(ticket.getId(), spicyFood.getId());
+
+        getTicketConsumptions(ticket).
+            then().
+                statusCode(HttpStatus.SC_OK).
+                body("$", hasSize(1)).
+                body("name", containsInAnyOrder(spicyFood.getName()));
+    }
+
+    @Test
+    public void getTicketConsumptions_OneConsumedAndDeleted(){
```

Comment:
This one actually threw an error during development which made me find a bug where a ticket would have two ConsumptionMaps. Because of this I'd like to keep it here. It has proven itself!